### PR TITLE
OHRM5X-2109: Validate enabled modules

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+class MyInfoAPI extends Endpoint implements ResourceEndpoint
+{
+    use AuthUserTrait;
+    use EmployeeServiceTrait;
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
```
Comment (ManulMax) [inline src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php:41]: Need to add API doc and testing

File: src/plugins/orangehrmPimPlugin/config/routes.yaml [hunk 2]
```diff
@@ -508,6 +508,14 @@ apiv2_pim_validate_employee_other_email:
   requirements:
     empNumber: '\d+'
 
+apiv2_pim_my_info:
+  path: /api/v2/pim/my-info
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPimPlugin/config/routes.yaml:512]: can go for `/api/v2/pim/myself` or `/api/v2/pim/me` :thinking: 
ref: https://stackoverflow.com/questions/36520372/designing-uri-for-current-logged-in-user-in-rest-applications

File: src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        $empNumber =  $this->getAuthUser()->getEmpNumber();
+        $employee = $this->getEmployeeService()->getEmployeeByEmpNumber($empNumber);
+        $this->throwRecordNotFoundExceptionIfNotExist($employee, Employee::class);
+
+        return new EndpointResourceResult(EmployeePersonalDetailModel::class, $employee);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php:47]:
```suggestion
        return new EndpointResourceResult(EmployeeDetailedModel::class, $employee);
```
:thinking: 
or we can go for multi model support,
`default` => EmployeeModel
`detailed` => EmployeeDetailedModel

**Review by RajithaKumara — CHANGES_REQUESTED** [review]
Pending:

- API doc
- Testing

File: src/plugins/orangehrmPimPlugin/config/routes.yaml (hunk 2 above)
Comment (ManulMax) [reply src/plugins/orangehrmPimPlugin/config/routes.yaml]: Changed to /api/v2/pim/myself

File: src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php (hunk 3 above)
Comment (ManulMax) [reply src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php]: add multi model support

File: src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+     *     path="/api/v2/pim/myself",
+     *     tags={"Pim/My Self"},
+     *     @OA\Parameter(
+     *         name="model",
+     *         in="query",
+     *         required=false,
+     *         @OA\Schema(
+     *             type="string",
+     *             enum={OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DEFAULT, OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DETAILED},
+     *             default=OrangeHRM\Pim\Api\MyInfoAPI::MODEL_DEFAULT
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPimPlugin/Api/MyInfoAPI.php:63]:
```suggestion
     *             enum={MyInfoAPI::MODEL_DEFAULT, MyInfoAPI::MODEL_DETAILED},
     *             default=MyInfoAPI::MODEL_DEFAULT
```
:thinking:

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**